tests/Controller/EnMarche/CitizenProjectControllerTest.php

```diff
@@ -200,6 +202,82 @@ public function testCommitteeSupportCitizenProject()
         ), trim($flash->text()));
     }
 
+    public function testAnonymousUserIsNotAllowedToFollowCitizenProject()
+    {
+        $committeeUrl = sprintf('/projets-citoyens/%s', 'le-projet-citoyen-a-paris-8');
+
+        $crawler = $this->client->request(Request::METHOD_GET, $committeeUrl);
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+        $this->assertFalse($this->seeFollowLink($crawler));
+        $this->assertFalse($this->seeUnfollowLink($crawler));
+        $this->assertTrue($this->seeRegisterLink($crawler));
+    }
+
+    public function testAuthenticatedCitizenProjectAdministratorCanUnfollowCommittee()
+    {
+        // Login as administrator
+        $this->authenticateAsAdherent($this->client, 'jacques.picard@en-marche.fr', 'changeme1337');
+        $crawler = $this->client->request(Request::METHOD_GET, sprintf('/projets-citoyens/%s', 'le-projet-citoyen-a-paris-8'));
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+
+        $unfollowButton = $crawler->filter('.citizen-project-unfollow');
+
+        // Button should be disabled for adminitrator
+        $this->assertNotNull($unfollowButton->attr('disabled'));
+        $this->assertSame('En tant que porteur de projet citoyen, vous ne pouvez pas cesser de le suivre.', trim($crawler->filter('.citizen-project-follow--anonymous__link')->text()));
+    }
+
+    public function testAuthenticatedAdherentCanFollowCitizenProject()
+    {
+        $this->authenticateAsAdherent($this->client, 'benjyd@aol.com', 'HipHipHip');
+
+        // Browse to the citizen project details page
+        $citizenProjectUrl = sprintf('/projets-citoyens/%s', 'le-projet-citoyen-a-paris-8');
+
+        $crawler = $this->client->request(Request::METHOD_GET, $citizenProjectUrl);
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+        $this->assertSame('2 acteurs', $crawler->filter('#followers > h3')->text());
+        $this->assertTrue($this->seeFollowLink($crawler));
+        $this->assertFalse($this->seeUnfollowLink($crawler));
+        $this->assertFalse($this->seeRegisterLink($crawler, 0));
+
+        // Emulate POST request to follow the committee.
```

Comment:
Perhaps we should consider using a client that can interpret javascript one day.

Reply:
YES !!! behat and selenium ❤️